### ISSUE TITLE
error while running fastml

### ISSUE THREAD

**Issue by ghost** [Team-fastML/fastML#27]
hello,
I am running fastml to develop phylogenetic tree but its showing error and no output is being generated.
The command use is perl FastML_Wrapper.pl --MSA_File PF05505_EBOLA_pfam.txt --seqType AA --outdir /home/nsm/k/kx/kxj190026/molevol/
The error:
outDir: /home/nsm/k/kx/kxj190026/molevol/
SubMatrix=JTT (default)
Copy and analyse MSA: /home/nsm/k/kx/kxj190026/molevol/PF05505_EBOLA_pfam.txt
LOG: /home/nsm/k/kx/kxj190026/molevol/FastML_log.log
cd /home/nsm/k/kx/kxj190026/molevol/; /home/nsm/k/kx/kxj190026/molevol/FastML.v3.11/www/fastml/../../programs/fastml/fastml -s /home/nsm/k/kx/kxj190026/molevol/PF05505_EBOLA_pfam.txt -mj -qf -g > /home/nsm/k/kx/kxj190026/molevol/fastml.std
--- Iter=0 logL=-5140.31
LikelihoodLast was not sent to bblEM
--- Iter=1 logL=-5132.6
LikelihoodLast was not sent to bblEM
--- Iter=2 logL=-5128.84
LikelihoodLast was not sent to bblEM
--- Iter=3 logL=-5126.4
LikelihoodLast was not sent to bblEM
--- Iter=4 logL=-5124.6
LikelihoodLast was not sent to bblEM
--- Iter=5 logL=-5123.19
LikelihoodLast was not sent to bblEM
--- Iter=6 logL=-5122.05
LikelihoodLast was not sent to bblEM
--- Iter=7 logL=-5121.13
LikelihoodLast was not sent to bblEM
--- Iter=8 logL=-5120.36
LikelihoodLast was not sent to bblEM
--- Iter=9 logL=-5119.73
LikelihoodLast was not sent to bblEM
--- Iter=0 logL=-5115.69
LikelihoodLast was not sent to bblEM
--- Iter=1 logL=-5114.06
LikelihoodLast was not sent to bblEM
--- Iter=2 logL=-5112.98
LikelihoodLast was not sent to bblEM
--- Iter=3 logL=-5112.16
LikelihoodLast was not sent to bblEM
--- Iter=4 logL=-5111.52
LikelihoodLast was not sent to bblEM
--- Iter=5 logL=-5111
LikelihoodLast was not sent to bblEM
--- Iter=6 logL=-5110.57
LikelihoodLast was not sent to bblEM
--- Iter=7 logL=-5110.21
LikelihoodLast was not sent to bblEM
--- Iter=8 logL=-5109.92
LikelihoodLast was not sent to bblEM
--- Iter=9 logL=-5109.68
LikelihoodLast was not sent to bblEM
--- Iter=0 logL=-5108.26
LikelihoodLast was not sent to bblEM
--- Iter=1 logL=-5107.66
LikelihoodLast was not sent to bblEM
--- Iter=2 logL=-5107.23
LikelihoodLast was not sent to bblEM
--- Iter=3 logL=-5106.91
LikelihoodLast was not sent to bblEM
--- Iter=4 logL=-5106.65
LikelihoodLast was not sent to bblEM
--- Iter=5 logL=-5106.44
LikelihoodLast was not sent to bblEM
--- Iter=6 logL=-5106.26
LikelihoodLast was not sent to bblEM
--- Iter=7 logL=-5106.12
LikelihoodLast was not sent to bblEM
--- Iter=8 logL=-5106
LikelihoodLast was not sent to bblEM
--- Iter=9 logL=-5105.91
LikelihoodLast was not sent to bblEM
--- Iter=0 logL=-5105.29
LikelihoodLast was not sent to bblEM
--- Iter=1 logL=-5105.02
LikelihoodLast was not sent to bblEM
--- Iter=2 logL=-5104.84
LikelihoodLast was not sent to bblEM
--- Iter=3 logL=-5104.69
LikelihoodLast was not sent to bblEM
--- Iter=4 logL=-5104.58
LikelihoodLast was not sent to bblEM
--- Iter=5 logL=-5104.49
LikelihoodLast was not sent to bblEM
--- Iter=6 logL=-5104.41
LikelihoodLast was not sent to bblEM
--- Iter=7 logL=-5104.35
LikelihoodLast was not sent to bblEM
--- Iter=8 logL=-5104.3
LikelihoodLast was not sent to bblEM
--- Iter=0 logL=-5104.11
LikelihoodLast was not sent to bblEM
--- Iter=1 logL=-5103.99
LikelihoodLast was not sent to bblEM
--- Iter=2 logL=-5103.9
LikelihoodLast was not sent to bblEM
--- Iter=3 logL=-5103.83
LikelihoodLast was not sent to bblEM
--- Iter=4 logL=-5103.77
LikelihoodLast was not sent to bblEM
--- Iter=5 logL=-5103.73
LikelihoodLast was not sent to bblEM
cd /home/nsm/k/kx/kxj190026/molevol/; perl /home/nsm/k/kx/kxj190026/molevol/FastML.v3.11/www/fastml/IndelReconstruction_Wrapper.pl --MSA_File /home/nsm/k/kx/kxj190026/molevol/PF05505_EBOLA_pfam.txt --Tree_File /home/nsm/k/kx/kxj190026/molevol/tree.newick.txt --outDir /home/nsm/k/kx/kxj190026/molevol/ --seqType aa --indelCutOff 0.5  > /home/nsm/k/kx/kxj190026/molevol//IndelReconstruction.log
Seq /home/nsm/k/kx/kxj190026/molevol/PF05505_EBOLA_pfam.txt
numberOfSeqs=4
seqLen=716

Step (1) readSequenceIntoGaps...
 All MSA gaps are sorted by coordinates
 Sort by: I =(i1,i2), K =(k1,k2), I<K iff i1<k1 or i1=k1 and i2<k2

There are 20 gaps

Step (2) delimitationOfCharacters... Simple Coding (SIC)
 Finding the required positions(=characters) in the coded sequence
 The number of characters <= the number of found gaps (each character may include several gaps)

There were 20 characters

Step (3) determinationCharacterState...
WARN: compute correlatins for co-Loss, printComputedCorrelationsData() is problematic (not all pair will have both co-gain and co-loss defined)
WARN: compute correlatins for co-Loss, printComputedCorrelationsData() is problematic (not all pair will have both co-gain and co-loss defined)
+=============================================================+
+  The gainLoss project:
+  Analysis of Phyletic Patterns in a Likelihood Framework
+  Version: gainLoss.VR01.266 - last updated 14.10.2013
+  Ofir Cohen - ofircohe@tau.ac.il
+=============================================================+

 ---------------------- THE PARAMETERS ----------------------------
Likelihood computation is performed while acounting for un-oberved data
With min number of presences('1's)= 1
With min number of absences('0's)= 0
'Reversible'(Root.freq==stationary.freq) model is used
RootFreq EQ stationary (taken from each sp - gain/(gain+loss) )
inTree file: /home/nsm/k/kx/kxj190026/molevol/tree.newick.txt.ForIndelReconstruction
inSeq file: /home/nsm/k/kx/kxj190026/molevol/IndelsReconstruction//PF05505_EBOLA_pfam.indelOutputFastaFile
log: RESULTS//log.txt with level= 5
Optimization of the model parmeters is performed
rate inference method is: empirical Bayesian estimate
using a Gamma prior distribution with: 4 discrete categories
branch lengths are not optimized
rate4site is calculated
calculePosteriorExpectationOfChange is done Analytically
-------------------------------------------------------------------

 startSequenceContainer
numberOfSeqs=4
seqLen=20
Num of zeros=60
Num of ones=36
Num of unKnowns=16

 startStochasticProcess...

InitGainLossByEmpiricalFreq: freq 1=init_gain= 0.3125
rateDist GAMMA with: initAlphaRate=0.5 and _numberOfRateCategories= 4
Stochastic process normalized with norm_factor=0.429688
startStochasticProcess SimpleGamma of 4 categories...
with gain=0.3125 loss=0.6875 root(1)=0.2
Stochastic process 'simple' normalized with norm_factor=0.429688

 startingEvolTreeTopology...
default rooting used, root name is N1
sons of root are:
S2  S4  N2
unObservableData object initialized with number of unObservable patterns= 1
The Tree Likelihood AllPosAlphTheSame is -37.721
Starting normalizeQandTree...(so that sumQii=1 (or weighted ave. of sunOii's for many Qs))
Q were multiplied by 1
The Tree Likelihood AllPosAlphTheSame is -37.721
 AlphaRate 0.5
 Gain 0.727273
 Loss 1.6
        Gain/Loss  ratio= 0.454545
 Stationary '1' Freq at the root (g/(g+l) = 0.3125
 Total branch lengths:5.40937

Starting MaxParsimonyChange...
 MaxParsimony with costMatrix - gainLossRatio 1:2
Cost of tree is 32 (with 20 events)
 Gain=12
 Losses=8
TIME = 0.0166667 minutes

 *** Start Optimizations

 WARN: no branch length estimation is performed with too few positions =20

 WARN: Skip MultipleAllBranchesByFactorAtStart with too few number of positions 20
  multipleAllBranchesByFactorAtStartByMaxParsimonyCost Total branch lengths: 1 with respect to costOfTreeMP 20
-------------------------------
Starting optimizations: maxNumIterations=1

######### optimization starting epsilonCycle=5 maxNumIterations=1
start optimization with L= -48.5956

------ Model+BBL iter=1
### optimization starting- epsilonOptParam=0.04999 epsilonOptIter= 0.14997, MaxNumIterations=1
start optimization with:
 L= -48.5956 gainLossRatio= 0.454545 gain= 0.727273 loss= 1.6 Alpha= 0.5

---- iter=1
currBestL= -29.3531     GainLossRatio= 0.0125557
currBestL= -27.9035     Alpha= 94.0399
 Too many iterations in optimizeGainLossModel. Iter= 2 Last optimized parameters are used. iter=2
-------------------------------
Model optimization over with:
Gain 0.112052
Loss 40.3226
        Gain/Loss  ratio= 0.00277889
AlphaRateRate 94.0399
updated likelihood (after optimizeParameters)= -27.9035
Model optimization RUNNING TIME = 0 minutes

 Too many=1 iterations in Model+BBL. Last optimized parameters are used.
Optimization RUNNING TIME = 0 minutes

# Best likelihood after optimization=-27.9035
TIME = 0 minutes

Starting normalizeQandTree...(so that sumQii=1 (or weighted ave. of sunOii's for many Qs))
Q were multiplied by 1
Tree branches multi by 1
The Tree Likelihood AllPosAlphTheSame is -27.9035
 AlphaRate 94.0399
 Gain 0.506278
 Loss 40.3226
        Gain/Loss  ratio= 0.0125557
 Stationary '1' Freq at the root (g/(g+l) = 0.0124
 Total branch lengths:1

 WARN: distanceFromRootForRecent=1, distanceFromNearestOTUForRecent=0.000001 are not estimated
Starting printLofPos...

Starting rate4site...
Running rate4site...
perform computeEB_EXP_siteSpecificRate... while computing posteriorProb PerCategory PerPosition
computeRate4site RUNNING TIME = 0 minutes
TIME = 0 minutes

Starting calculePosteriorExpectationOfChange...

Computation stochastic mapping
Computation performed analytically for rate 0.872149
Computation performed analytically for rate 0.963545
Computation performed analytically for rate 1.03039
Computation performed analytically for rate 1.13391
TIME = 0 minutes

Mean   values Gain=1.44782      Loss=0.491302
Median values Gain=1.44974      Loss=0.489805

TIME = 0 minutes (calculePosteriorExpectationOfChange)

Starting ancestralReconstructorBasedOnJoint...
computeAncestralPosterior (take into acount joint probabilty)...
TIME = 0 minutes

I don't know whether its correct. Since in the beginning its showing error as likelihood last was not sent to bblem
